sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/json/JsonInferSchema.scala

```diff
@@ -81,7 +81,8 @@ class JsonInferSchema(options: JSONOptions) extends Serializable with Logging {
    */
   def infer[T](
       json: RDD[T],
-      createParser: (JsonFactory, T) => JsonParser): StructType = {
+      createParser: (JsonFactory, T) => JsonParser,
+      isReadFile: Boolean = true): StructType = {
```

Comment:
Do you really need the default value. It is called just from 2 places, and from 2 places in tests where you set it wrongly actually because of the default value.

Reply:
just in case this is called by external code, as this is not `private[sql]`.

Reply:
OK maybe the default should be false, to match the behavior before #42979